modules/avformat/avformat.c

```diff
@@ -206,7 +208,19 @@ int avformat_shared_alloc(struct shared **shp, const char *dev)
 	if (err)
 		goto out;
 
-	ret = avformat_open_input(&st->ic, dev, NULL, NULL);
+	input_format = av_find_input_format(dev);
+
+	if (cfg->video.width) {
+		re_snprintf(buf, sizeof(buf), "%dx%d",
+			    cfg->video.width, cfg->video.height);
```

Comment:
the input resolution and framerate should be taken from vidsrc input arguments:

```c
int avformat_video_alloc(struct vidsrc_st **stp, const struct vidsrc *vs,
			 struct media_ctx **ctx, struct vidsrc_prm *prm,
			 const struct vidsz *size, const char *fmt,
			 const char *dev, vidsrc_frame_h *frameh,
			 vidsrc_error_h *errorh, void *arg)
```

you need to pass `prm->fps` and `size` into this function:

```c
		err = avformat_shared_alloc(&st->shared, dev);
		if (err)
			goto out;
```

and also a flag to say if the shared is opened by audio or video source.

only in the case of opening the shared from video, should you set
resolution and framerate.

if the shared context is opened by audio source (e.g when using a mp4 mediafile),
should the wanted resolution not be set.

Reply:
Should now be done.  Couple of observations:

1. Setting of video_fps in config does not seem to have any effect when fps is taken from struct vidsrc_prm.  May be the same for video_size (haven't tried yet).

2. What to do with unused variable resulting from removing of `if (now < (offset + xts))`?